my_list.rb

```diff
@@ -0,0 +1,28 @@
+require_relative 'my_enumerable'
+class MyList
+  def initialize(*list)
+    @list = list
+  end
+
+  attr_reader :list
+
+  def each
+    yield @list
+  end
```

Comment:
- [ ] You did everything well except your implementation of the `each` method. When I try to test the `each` method, I ran the code snippet below 👇🏾 

```ruby
my_list_obj.each { |e| puts"#{e}: check"} # => [1, 2, 3, 4]: check
```

The code result was -> `[1, 2, 3, 4]: check` instead of listing each item as shown below 👇🏾 
```ruby
my_list_obj.each { |e| puts"#{e}: check"} # => 
# 1: check
# 2: check
# 3: check
# 4: check

def each
    return to_enum(:each) unless block_given?

    counter = 0
    while counter < @list.length
      yield(@list[counter])
      counter += 1
    end
    @list
end
```

The above is the implementation of the `each` method. Kindly use this as a guide to implementing your `each` correctly.